### PR TITLE
Supported subtitle formats and file extensions

### PR DESCRIPTION
**Pull request by porg**
- Extending manual section "Subtitles".
- Fulfills https://github.com/mpv-player/mpv.io/issues/104

### CODE REVIEW

**Comment by Dudemanguy** [comment]
I think simply a link to ffmpeg documentation is sufficient here personally.

File: DOCS/man/options.rst [hunk 1]
```diff
@@ -2241,6 +2241,8 @@ Audio
 Subtitles
 ---------
 
+.. note::
+
```
Comment (Dudemanguy) [inline DOCS/man/options.rst:2245]: Don't need this.

**Comment by porg** [comment]
I think that extra contextualization infos, like pointing users to look for the "Decoding" column, helps "consumer level users" who possibly have no idea what's the difference between Muxing / Demuxing / Encoding / Decoding.

**Comment by porg** [comment]
Why should users of a video player like mpv (often pure consumer level users) know the terminology from video encoding/transcoding software such as ffmpeg?

**Comment by Dudemanguy** [comment]
Well sure a brief sentence saying that it's the decoding column that's relevant is fine, but I was talking about the rest of it. No real need to try and list out the subtitle extensions in the manpage and all that. That's just noise imo. The better solution would be to add help output to --audio/video/sub-auto.

**Comment by porg** [comment]
Ideal position within the online manual:
- Ofc am open for discussion!
- Your proposed "--audio/video/sub-auto" section I could not find literally nor by interpretation within the manual.
- To me the correct place is somewhere in the "Subtitle" section, at best right at its start because "supported file formats and the supported file extensions" concerns the whole section. Ideally I had wanted this to be a sub-heading with an anchor one can link to. But nowhere else in the manual this style was used, and I wanted to adhere to the overall style/structure, hence I integrated it into the "Note" box at the beginning of the "Subtitles" section.

**Comment by Dudemanguy** [comment]
My help idea was a bad one (there's already help output for those options duh). A command line option to print out supported extensions might not be bad sort of like how there's already --list-options, --list-protocols, etc. Of course, this requires code changes.

Anyways as for this proposal, I don't agree with having a list of extensions that could go out of date at any time and needs to be manually kept up with in the manual. Giving information to users is valuable, but it should be done in a correct way.

**Comment by porg** [comment]
The automated way would be to integrate the value of sub_ext[] or scrape the ffmpeg support matrix or a combination of both.

Maybe you have that skills?

If that ain’t possible I still do not see why putting the supported formats of 2023 there (those have accumulated over 10+ years certainly) for SEO instead of nothing at all!

So many new subtitle formats won’t come anymore I assume. I estimate that we are already past the development peak in that domain as video/subtitles are already a pretty ripe technology. So a new exotic extra format every 9-24 months. No bummer if that’s not added immediately IMHO. 

Not mentioning anything is really the worst choice in that subject matter IMHO.

Poll: Thumbs up/down.

**Comment by Dudemanguy** [comment]
It's not a democracy. If any other developer wants to chime in, they are of course welcome to, but clearly this isn't going anywhere so I won't spend anymore energy on it.

**Comment by christoph-heinrich** [comment]
I'm not against a simple `supported subtitle formats can be fount at <ffmpeg url> in the decoding column`, but more then that seems unnecessary.

**Comment by porg** [comment]
> It's not a democracy.

That's clear. Thats why I called it a poll and not a vote. And anyhow the qualitative comments such as @christoph-heinrich 's are more interesting than up/down-votes.

So at least a minimal addition please.

If you don't want the concrete mentions of the file formats/extensions, then the downside is you loose a SEO potential, which is acceptable.

Please add least add the minimal consensus text. Thanks!

**Pull request closed without merging**